pallets/pallet-validator/src/lib.rs

```diff
@@ -0,0 +1,334 @@
+		/// A user can only have 1 validator request at a time
+		#[pallet::weight(100_000_000)]
+		pub fn create_validator(
+			origin: OriginFor<T>,
+			validator_node_account: T::AccountId,
+			stash_account: T::AccountId,
+			description: Vec<u8>,
+			tf_connect_id: u64,
+			info: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin.clone())?;
+
+			// Request should not be a duplicate
+			ensure!(
+				!<Validator<T>>::contains_key(&address),
+				Error::<T>::DuplicateValidator
+			);
+
+			// Request stash account should be bonded
+			Self::check_bond(&stash_account, &address)?;
+
+			let request = types::Validator {
+				validator_node_account: validator_node_account.clone(),
+				stash_account,
+				description,
+				tf_connect_id,
+				info,
+				state: types::ValidatorRequestState::Created,
+			};
+
+			// Create a validator request object
+			<Validator<T>>::insert(&address, &request);
+
+			Self::deposit_event(Event::ValidatorCreated(address, request.clone()));
+
+			Ok(().into())
+		}
+
+		/// Start participating in consensus
+		/// Will activate the Validator node account on consensus level
+		/// A user can only call this if his request to be a validator is approved by the council
+		/// Should be called when his node is synced and ready to start validating
+		#[pallet::weight(100_000_000)]
+		pub fn activate_validator(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin)?;
+
+			let mut validator = <Validator<T>>::get(&address)
+				.ok_or(DispatchError::from(Error::<T>::ValidatorNotFound))?;
+
+			ensure!(
+				validator.state != types::ValidatorRequestState::Validating,
+				Error::<T>::ValidatorValidatingAlready
+			);
+			ensure!(
+				validator.state == types::ValidatorRequestState::Approved,
+				Error::<T>::ValidatorNotApproved
+			);
+
+			// Update the validator request
+			validator.state = types::ValidatorRequestState::Validating;
+			<Validator<T>>::insert(address, &validator);
+
+			// Add the validator and rotate
+			substrate_validator_set::Pallet::<T>::add_validator(
+				frame_system::RawOrigin::Root.into(),
+				validator.validator_node_account.clone(),
+			)?;
+
+			Self::deposit_event(Event::ValidatorActivated(validator));
+
+			Ok(().into())
+		}
+
+		/// Change node validator account
+		/// In case the Validator wishes to change his validator node account
+		/// he can call this method with the new node validator account
+		/// this new account will be added as a new consensus validator
+		#[pallet::weight(100_000_000)]
+		pub fn change_node_validator_account(
+			origin: OriginFor<T>,
+			new_node_validator_account: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin)?;
+
+			let mut validator = <Validator<T>>::get(&address)
+				.ok_or(DispatchError::from(Error::<T>::ValidatorNotFound))?;
+
+			ensure!(
+				validator.state == types::ValidatorRequestState::Validating,
```

Comment:
Why shouldn't it be possible to change the validator_node_account before it is activated?

Reply:
Because 

1. we don't know if he is validating or not and need to update consensus or not
2. this implies he needs to call activate again after changing his node validator